Added signbit() to cpp docs (#1053)
Codecademy/docs: content/cpp/concepts/math-functions/terms/signbit/signbit.md

---
Title: 'signbit()'

> Description: 'Returns true if the floating point is negative and false if floating point is positive.'

Reviewer: ```suggestion
Description: 'Returns true if the floating point argument is negative and false if floating point argument is positive.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The `signbit()` function returns true if the sign of the  number is negative and false if sign is positive.

Reviewer: ```suggestion
The **`signbit()`** function returns true if the sign of the argument is negative and false if its sign is positive.
```

> The **`signbit()`** function returns true if the sign of the  argument is negative and false if its sign is positive.

Reviewer: ```suggestion
The **`signbit()`** function returns true if the sign of the argument is negative and false if its sign is positive.
```

## Syntax

The `cmath` library must be added at the top of the file.

```cpp
std::signbit(n);
```

Argument must be a `double`/`float`/`long double`/`int`, and the return value will be the same type.
This function also detects the signs of zeroes, intfinities and NaNs.

## Example

```cpp
#include <iostream>
#include <cmath>

int main()
{

>     double a=9.0,b=0.0;

Reviewer: ```suggestion
    double a=9.0, b=0.0;
```

    // Here, c is infinity
    double c=a/b;
    // if c tends to positive infinity
    if(std::signbit(c)==false)
    {

>         std::cout<<"c tends to positive infinity";

Reviewer: ```suggestion
        std::cout << "c tends to positive infinity";
```

    }
    else
    {

>         std::cout<<"c tends to negative infinity";

Reviewer: ```suggestion
        std::cout << "c tends to negative infinity";
```

    }
    return 0;
}
```

## Codecademy Example

Use `signbit()` to know whether the integer is positive or negative:

```codebyte/cpp
#include <iostream>
#include <cmath>

> int main() {

Reviewer 2: ```suggestion
int main()
{
```

  double n = 12.3456;
  bool result;

  result = std::signbit(n);

  if(result)
  {

>     std::cout<<"n is negative";

Reviewer: ```suggestion
    std::cout << "n is negative";
```

  }
  else
  {

>     std::cout<<"n is positive";

Reviewer: ```suggestion
    std::cout << "n is positive";
```

---
Title: 'signbit()'
Description: 'Returns true if the floating point argument is negative and false if floating point argument is positive.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **`signbit()`** function returns `true` if the sign of the argument is negative and `false` if its sign is positive.

## Syntax

The `cmath` library must be added at the top of the file.

```pseudo
std::signbit(n);
```

> Argument `n` must be of type `double`/`float`/`long double`/`int`, and the return value will be of that type. This function also detects the signs of zeroes, infinities and `NaN`s.

Reviewer 2: ```suggestion
Argument `n` must be of type `double`/`float`/`long double`/`int`, and the return value will be of that type. This function also detects the signs of zeroes, infinities, and `NaN`s.
```

## Example

```cpp
#include <iostream>
#include <cmath>

int main()
{
    double a = 9.0, b = 0.0;
    // Here, c is infinity because a number divided by 0
    // evaluates to positive infinity when there is a limit.
    double c = a/b;

    // If c is false, print out that it tends to positive infinity.

>     if (std::signbit(c)==false)

Reviewer 2: ```suggestion
    if (std::signbit(c) == false)
```

    {
        std::cout << "c tends to positive infinity";
    }
    else
    {
        std::cout << "c tends to negative infinity";
    }
    return 0;
}
```

## Codecademy Example

> Use `signbit()` and outputs a string about whether the integer is positive or negative:

Reviewer 2: ```suggestion
The example below outputs a string about whether the integer is positive or negative:
```
